src/Microsoft.Identity.Web/CertificateManagement/DefaultCertificateLoader.cs

```diff
@@ -13,265 +13,265 @@
 
 namespace Microsoft.Identity.Web
 {
-    /// <summary>
-    /// Certificate Loader.
-    /// Only use when loading a certificate from a daemon application, or an ASP NET app, using MSAL .NET directly.
-    /// For an ASP NET Core app, <b>Microsoft Identity Web will handle the certificate loading</b> for you.
-    /// <example><code>
-    /// IConfidentialClientApplication app;
-    /// ICertificateLoader certificateLoader = new DefaultCertificateLoader();
-    ///     certificateLoader.LoadIfNeeded(config.CertificateDescription);
-    ///
-    ///    app = ConfidentialClientApplicationBuilder.Create(config.ClientId)
-    ///           .WithCertificate(config.CertificateDescription.Certificate)
-    ///           .WithAuthority(new Uri(config.Authority))
-    ///           .Build();
-    /// </code></example>
-    /// </summary>
-    public class DefaultCertificateLoader : ICertificateLoader
-    {
-        /// <summary>
-        /// User assigned managed identity client ID (as opposed to system assigned managed identity)
-        /// See https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal.
-        /// </summary>
-        public static string? UserAssignedManagedIdentityClientId { get; set; }
+	/// <summary>
+	/// Certificate Loader.
+	/// Only use when loading a certificate from a daemon application, or an ASP NET app, using MSAL .NET directly.
+	/// For an ASP NET Core app, <b>Microsoft Identity Web will handle the certificate loading</b> for you.
+	/// <example><code>
+	/// IConfidentialClientApplication app;
+	/// ICertificateLoader certificateLoader = new DefaultCertificateLoader();
+	///	 certificateLoader.LoadIfNeeded(config.CertificateDescription);
+	///
+	///	app = ConfidentialClientApplicationBuilder.Create(config.ClientId)
+	///		   .WithCertificate(config.CertificateDescription.Certificate)
+	///		   .WithAuthority(new Uri(config.Authority))
+	///		   .Build();
+	/// </code></example>
+	/// </summary>
+	public class DefaultCertificateLoader : ICertificateLoader
+	{
+		/// <summary>
+		/// User assigned managed identity client ID (as opposed to system assigned managed identity)
+		/// See https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal.
+		/// </summary>
+		public static string? UserAssignedManagedIdentityClientId { get; set; }
```

Comment:
Having this be static and externally settable opens up race conditions where someone thinks it is populated when it may not be, and then makes incorrect decisions based upon that. Saying it is not used that way now is not sufficient. This is a high risk of misuse for future contributors. Recommendation is to make it an instance field on a new interface that is not settable externally, then access it through dependency injection.

Reply:
will raise an issue for this.

Reply:
https://github.com/AzureAD/microsoft-identity-web/issues/1095